vignettes/baselinenowcast.Rmd

```diff
@@ -167,91 +145,39 @@ The reporting triangle will be used to estimate the delay distribution, or the p
-
-```{r}
-latest_training_data <- enw_latest_data(training_data)
-```
-
-To obtain the data we want to evaluate the forecasts against, we will use `enw_filter_reference_dates()` applied to the `target_data`, to filter it for only the `n_training_volume` days of historical data.
+Next, we will specify the amount of training volume used to fit the model as a function of the maximum delay, and the proportion of the total training volume used for delay estimation (with the remainder being used for uncertainty estimation).
+Internally within `baselinenowcast()` we will call, `allocate_reference_times()` with these arguments.
+We'll us the default settings for this function here, which uses 3 times the maximum delay for the total training volume with 50% used for delay estimation.
 
 ```{r}
-eval_data <- enw_filter_reference_dates(
-  latest_data,
-  include_days = n_training_volume
-)
-```
-
-Next we obtain the reporting triangle, adding an additional day to the `max_delay` because we want the `max_delay + 1` entries since 0 is a valid delay.
-This also validates that the data is in the correct format and runs preprocessing see `enw_preprocess_data()` for more details.
-
-```{r}
-pobs <- enw_preprocess_data(
-  obs = training_data,
-  max_delay = max_delay + 1
-)
+scale_factor <- 3
+prop_delay <- 0.5
 ```
 
-Extract components from `pobs`, which is an `epinowcast` object.
-We only have one group here so we only need `reference_date`, `delay`, and `new_confirm`.
+Next we obtain a `reporting_triangle` using the `as_reporting_triangle()` function, which expects a data.frame with case counts indexed by reference date and report date and the maximum delay.
+This function computes the delay between the reference and report date and pivots the data from long to wide, so that the rows are reference times and the columns indicate the delay between the reference and report date, and the entries indicate the incident case counts.
+This also validates that the data is in the correct format and runs pre-processing to fill in any missing dates, see `?as_reporting_triangle.data.frame` and `?reporting_triangle` for more details on required inputs and the format of the `reporting_triangle` object.
 
 ```{r}
-reporting_triangle_df <- select(
-  pobs$new_confirm[[1]],
-  reference_date,
-  delay,
-  new_confirm
+rep_tri <- as_reporting_triangle(observed_data,
+  max_delay = max_delay
 )
 ```
 
-We now pivot to wide format, dropping the `reference_date` column, and convert to a matrix this is the format that the low-level functions in `baselinenowcast` expect.
-
-```{r}
-reporting_triangle <- reporting_triangle_df |>
-  pivot_wider(names_from = delay, values_from = new_confirm) |>
-  select(-reference_date) |>
-  as.matrix()
-```
+The `reporting_triangle` object is a matrix with named rows for the reference dates corresponding to each row of the matrix.
```

Comment:
Suggests it depends on the refactor but the code use below suggests it doesn't?

Reply:
Ah yes, this is confusing. I didn't implement it in the code because it would fail but (still requires `rep_tri$reporting_triangle_matrix`) but I described it based on the refactor as assumed that would get merged first. Will leave this open to address one way or the other (most likely will change code)